lib/_stream_writable.js

```diff
@@ -273,13 +273,14 @@ Writable.prototype.write = function(chunk, encoding, cb) {
       cb = nop;
   }
 
-  let err;
   if (state.ending) {
-    err = new ERR_STREAM_WRITE_AFTER_END();
+    throw new ERR_STREAM_WRITE_AFTER_END();
```

Comment:
I'm not sure if it is viable to throw here because there are cases where `.end()` is called automatically and that is outside of the user control.

Reply:
but this is `write()`, not `end()`?

Reply:
yes, but without knowing when `end()` is called the user needs to check `state.ending` before every write to ensure the error is not thrown no?

Reply:
ah, true. So what is your suggestion? Remove it like `ERR_STREAM_DESTROYED`?

Reply:
This is not really something we should swallow... since the data will disappear without any error...

Reply:
They should probably check `stream.writable`.

Reply:
Not sure, keep it as is to reduce the breaking change? Yes I also don't like to throw on some cases and not on others.

Reply:
> Not sure, keep it as is to reduce the breaking change? 

That really makes things confusing. How about this. Throw on invalid argument errors and otherwise keep previous behavior?

Reply:
ERR_STREAM_ALREADY_FINISHED
ERR_STREAM_WRITE_AFTER_END
ERR_STREAM_DESTROYED

all require the user to at least check `stream.writable` before calling `stream.write()`, which  I believe is very uncommon out in the wild.

Reply:
> That really makes things confusing. How about this. Throw on invalid argument errors and otherwise keep previous behavior?

Yes, that what I was trying to say in my previous comment.